datastore/dsiface/dsiface_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dsiface
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/datastore"
+)
+
+func TestIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("integration tests skipped in short mode")
+	}
+
+	kind := fmt.Sprintf("dsiface_test_%d", time.Now().UnixNano())
+	ctx := context.Background()
+	c, err := datastore.NewClient(ctx, os.Getenv("DATASTORE_PROJECT_ID"))
```

Comment:
Sorry, I should have been more clear. Here, you _should_ test for DATASTORE_PROJECT_ID (but not the creds env var). See bqiface_test.go.

Reply:
OK that sounds good!